File: src/main/java/com/simulator/log/LogType.java
package com.simulator.log;

/**
 * Tipe-tipe log untuk Write-Ahead Log
 */
public enum LogType {
    BEGIN("Transaction begin"),
    PREPARE("Prepare phase"),
    COMMIT("Commit phase"),
    ABORT("Abort phase"),
    CHECKPOINT("Checkpoint"),
    HEARTBEAT("Node heartbeat");
    
    private final String description;
    
    LogType(String description) {
        this.description = description;
    }
    
    public String getDescription() {
        return description;
    }
    
    @Override
    public String toString() {
        return description;
    }
}

File: src/log/log_type.py
"""
Log Type Enum - Tipe-tipe log entry dalam WAL
"""

from enum import Enum


class LogType(Enum):
    """
    Enum untuk tipe log entry dalam Write-Ahead Log
    """
    
    # Transaction lifecycle logs
    TRANSACTION_START = "TRANSACTION_START"
    TRANSACTION_COMMIT = "TRANSACTION_COMMIT"
    TRANSACTION_ABORT = "TRANSACTION_ABORT"
    
    # 2PC logs
    PREPARE = "PREPARE"
    COMMIT = "COMMIT"
    ABORT = "ABORT"
    
    # Phase logs
    PREPARE_PHASE = "PREPARE_PHASE"
    COMMIT_PHASE = "COMMIT_PHASE"
    ABORT_PHASE = "ABORT_PHASE"
    
    # Execution logs
    LOCAL_EXECUTION = "LOCAL_EXECUTION"
    
    # Recovery logs
    RECOVERY_START = "RECOVERY_START"
    RECOVERY_COMPLETE = "RECOVERY_COMPLETE"
    
    # Checkpoint
    CHECKPOINT = "CHECKPOINT"